testflinger/__init__.py

```diff
@@ -153,4 +138,47 @@ def unhandled_exception(exc):
     return tf_app
 
 
+def setup_mongodb(application):
+    """
+    Setup mongodb connection if we have valid config data
+    Otherwise leave it empty, which means we are probably running unit tests
+    """
+    mongo_user = os.environ.get("MONGO_USER")
+    mongo_pass = os.environ.get("MONGO_PASSWORD")
+    mongo_db = os.environ.get("MONGO_DATABASE")
+    if not application.config.get("MONGO_URI") and not (
+        mongo_user and mongo_pass and mongo_db
+    ):
+        # We are probably running unit tests
+        return
+
+    if not application.config.get("MONGO_URI"):
+        application.config[
+            "MONGO_URI"
+        ] = f"mongodb://{mongo_user}:{mongo_pass}@mongo:27017/{mongo_db}"
+
+    mongo_client = PyMongo(
+        application,
+        uri=application.config["MONGO_URI"],
+        uuidRepresentation="standard",
+    )
+    application.db = mongo_client.db
+    # PWL application.db = mongo_client
+
+    # Initialize collections and indices in case they don't exist already
+    # Automatically expire jobs after 7 days if nothing runs them
+    application.db.jobs.create_index(
+        "created_at", expireAfterSeconds=7 * 24 * 60 * 60
```

Comment:
I guess `7 * 24 * 60 * 60` should be defined as a constant at the beginning of the file, because one may try to change it. So it's better to change it in one place instead of each line where this number is defined.

Reply:
updated this in https://github.com/plars/testflinger/commit/7a498752dabb85511d44f4a73db0bd12bdd8d36a